aggregator/src/datastore.rs

```diff
@@ -875,36 +885,91 @@ impl<C: Clock> Transaction<'_, C> {
                 ],
             )
             .await?
+            .map(|row| Self::client_report_from_row(vdaf, *task_id, *report_id, row))
+            .transpose()
+    }
+
+    #[cfg(feature = "test-util")]
+    pub async fn get_client_reports_for_task<const L: usize, A: vdaf::Aggregator<L>>(
+        &self,
+        vdaf: &A,
+        task_id: &TaskId,
+    ) -> Result<Vec<LeaderStoredReport<L, A>>, Error>
+    where
+        A::InputShare: PartialEq,
+        A::PublicShare: PartialEq,
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+    {
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT
+                client_reports.report_id,
+                client_reports.client_timestamp,
+                client_reports.extensions,
+                client_reports.public_share,
+                client_reports.leader_input_share,
+                client_reports.helper_encrypted_input_share
+            FROM client_reports
+            JOIN tasks ON tasks.id = client_reports.task_id
+            WHERE tasks.task_id = $1",
+            )
+            .await?;
+        self.tx
+            .query(&stmt, &[/* task_id */ &task_id.as_ref()])
+            .await?
+            .into_iter()
             .map(|row| {
-                let time = Time::from_naive_date_time(&row.get("client_timestamp"));
+                let report_id_bytes: [u8; ReportId::LEN] = row
+                    .get::<_, Vec<u8>>("report_id")
+                    .try_into()
+                    .map_err(|err| {
+                        Error::DbState(format!("couldn't convert report_id value: {err:?}"))
+                    })?;
```

Comment:
nit: I think you could use `RowExt::get_bytea_and_convert` here, but you might need some more `From` or `TryFrom` implementations on `ReportId`, so it might not be worth the trouble.

Reply:
Sure, that's a little cleaner & it's a pattern repeated in a few places.